src/Metadata/Driver/EnumPropertiesDriver.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer\Metadata\Driver;
+
+use JMS\Serializer\Metadata\ClassMetadata as SerializerClassMetadata;
+use JMS\Serializer\Metadata\ExpressionPropertyMetadata;
+use JMS\Serializer\Metadata\PropertyMetadata;
+use JMS\Serializer\Metadata\StaticPropertyMetadata;
+use JMS\Serializer\Metadata\VirtualPropertyMetadata;
+use Metadata\ClassMetadata;
+use Metadata\Driver\DriverInterface;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionProperty;
+
+class EnumPropertiesDriver implements DriverInterface
+{
+    /**
+     * @var DriverInterface
+     */
+    protected $delegate;
+
+    public function __construct(DriverInterface $delegate)
+    {
+        $this->delegate = $delegate;
+    }
+
+    public function loadMetadataForClass(ReflectionClass $class): ?ClassMetadata
+    {
+        $classMetadata = $this->delegate->loadMetadataForClass($class);
+        \assert($classMetadata instanceof SerializerClassMetadata);
+
+        if (null === $classMetadata) {
+            return null;
+        }
+
+        // We base our scan on the internal driver's property list so that we
+        // respect any internal white/blacklisting like in the AnnotationDriver
+        foreach ($classMetadata->propertyMetadata as $propertyMetadata) {
+            // If the inner driver provides a type, don't guess anymore.
+            if ($propertyMetadata->type || $this->isVirtualProperty($propertyMetadata)) {
+                continue;
+            }
+
+            try {
+                $propertyReflection = $this->getReflection($propertyMetadata);
+                if ($enum = $this->getEnumReflection($propertyReflection)) {
+                    $serializerType = ['name' => 'enum', 'params' => [$enum->getName(), $enum->isBacked() ? 'value' : 'name']];
```

Comment:
May it would be better to change the type on ClassMetadata? At the moment it will create inconsistency between handling Enums in objects and arrays. 
Object:
```json
{
    "ordinary_auto_detect":"Clubs"
}
```
Array of Enums:
```json
[{"name":"Clubs"}]
```
Single Enum:
```json
{"name":"Clubs"}
```

Reply:
good catch! that is not easy to solve at all :(

Reply:
i have solved this with a subscriber